books/free-programming-books-subjects.md

```diff
@@ -99,6 +99,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Analysis and Design of Algorithms](http://www.cse.iitd.ernet.in/~ssen/csl356/admin356.html) - Sandeep Sen, IIT Delhi
 * [Animated Algorithm and Data Structure Visualization](http://visualgo.net) (Resource)
 * [Annotated Algorithms in Python: Applications in Physics, Biology, and Finance](https://github.com/mdipierro/nlib) - Massimo di Pierro
+* [Artificial Intelligence - Agents and Environments](http://www.e-booksdirectory.com/details.php?ebook=4457) - William John Teahan
```

Comment:
Is this an authorized distribution?

Reply:
@shrn01 Yes, this is an authorized distribution as per my findings.

Reply:
E-books directory is an aggregator site. notice that the links on the page you've supplied points to the Bookboon site. that means that we can't check to see if the book is still available. So we won't use links to ebook directory.
In other words, _it's not even a distribution_, let alone authorized!